### PR TITLE
[CHG] core: mark a bunch of utilities as deprecated

### PR DESCRIPTION
**Pull request by xmo-odoo**
Remove UnquoteEvalContext directly because it's unused and super specialised

### CODE REVIEW

File: odoo/tools/misc.py [hunk 1]
```diff
@@ -124,16 +136,17 @@ def exec_pg_environ():
...
     prog = find_pg_tool(name)
     env = exec_pg_environ()
-    with open(os.devnull) as dn:
-        args2 = (prog,) + args
-        rc = subprocess.call(args2, env=env, stdout=dn, stderr=subprocess.STDOUT)
-        if rc:
-            raise Exception('Postgres subprocess %s error %s' % (args2, rc))
+    subprocess.run((prog, *args), env=env, stdout=subprocess.DEVNULL, stderr=subprocess.STDOUT, check=True)
 
 def exec_pg_command_pipe(name, *args):
```
Comment (xmo-odoo) [inline odoo/tools/misc.py:141]: Maybe also deprecate the two `exec_pg_command`?

They don't do much, useful, they're only used 3 times in all, and one of these uses doesn't even seem correct:
```python
        if odoo.tools.exec_pg_command(pg_cmd, *pg_args):
            raise Exception("Couldn't restore database")
```
that condition and specialised error messages haven't done anything useful since ec9a543014a91be17398ba6e443578fe37c3a885

File: odoo/tools/misc.py [hunk 2]
```diff
@@ -1256,10 +1251,8 @@ def __lt__(self, other): return self.val > other.val
 
 @contextmanager
 def ignore(*exc):
-    try:
-        yield
-    except exc:
-        pass
+    warnings.warn("Since 16.0 `odoo.tools.ignore` is replaced by `contextlib.suppress`.", DeprecationWarning, stacklevel=2)
+    return contextlib.suppress(*exc)
```
Comment (xmo-odoo) [inline odoo/tools/misc.py:1255]: Maybe don't do this here as there are 6 callsites to remove?

File: odoo/tools/misc.py [hunk 3]
```diff
@@ -1572,13 +1564,7 @@ def format_duration(value):
...
     return '%02d:%02d' % (hours, minutes)
 
-
-def _consteq(str1, str2):
-    """ Constant-time string comparison. Suitable to compare bytestrings of fixed,
-        known length only, because length difference is optimized. """
-    return len(str1) == len(str2) and sum(ord(x)^ord(y) for x, y in zip(str1, str2)) == 0
-
-consteq = getattr(passlib.utils, 'consteq', _consteq)
+consteq = hmac_lib.compare_digest
```
Comment (xmo-odoo) [inline odoo/tools/misc.py:1567]: Deprecate as well? But it's used ~25 times. Maybe put ignore and consteq in separate PRs which fix their callsites?

Comment (xmo-odoo) [reply odoo/tools/misc.py:1567]: Also if deprecating, suggest `secrets.compare_digest`, it's the exact same function but avoids flagging about `hmac`.

Comment (Julien00859) [reply odoo/tools/misc.py:1560]: Maybe don't create another PR but just deprecate both ignore and consteq (+ fix their call sites) in dedicated commits in this PR

File: odoo/tools/misc.py [hunk 4]
```diff
@@ -1659,6 +1634,7 @@ def handle_style(html_diff, custom_style):
     return handle_style(diff, custom_style)
 
 
+# remove from general utils and move to the sole location where it's used?
 def traverse_containers(val, type_):
```
Comment (Julien00859) [inline odoo/tools/misc.py:1642]: +1 with the comment

**Review by Julien00859 — APPROVED** [review]

**Comment by xmo-odoo** [comment]
@robodoo override=ci/security
@robodoo override=ci/style

**Review by ryv-odoo — APPROVED** [review]

**Comment by xmo-odoo** [comment]
@rco-odoo what do you think of `_traverse_container`, should it remain an internal layer of the RPC layer, or should it be a more specific feature of `lazy` itself?

In the latter case not sure how to expose it, a method could conflict with the underlying object. Though I guess using a `_method` is always an option.

Also I'm realising that `traverse_container` is not perfect as it won't work in the case of a recursive lazy (e.g. a lazy resolving to an object which itself contains a lazy), but we're not using lazy *that much* atm so~~

An interesting alternative I was thinking about is instead of traversing (potentially large) collections looking for lazy objects, we could register lazy objects in a threadlocal / requestlocal weakset. This would increase the cost of lazy objects a bit, but make the end-of-request cheaper (as it would usually be iterating an empty weakset).

**Comment by xmo-odoo** [comment]
@robodoo r+

**Comment by xmo-odoo** [comment]
@robodoo rebase-ff
